src/WebJobs.Script/ExtensionBundle/ExtensionBundleManager.cs

```diff
@@ -259,7 +264,29 @@ private static string FindBestVersionMatch(VersionRange versionRange, IEnumerabl
                 return version;
             }).Where(v => v != null);
 
-            return bundleVersions.OrderByDescending(version => version.Version).FirstOrDefault()?.ToString();
+            var matchingVersion = bundleVersions.OrderByDescending(version => version.Version).FirstOrDefault();
+
+            if (bundleId != ScriptConstants.DefaultExtensionBundleId)
+            {
+                return matchingVersion?.ToString();
+            }
+
+            var maximumBundleV3Version = NuGetVersion.Parse(configOption.MaximumSupportedBundleV3Version);
```

Comment:
As part of option configuration, you could do the `NuGetVersion` parsing on the options object, and not need to do it here.